The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16727755158/job/47347895617) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47347895617) [(plain)](https://github.com/rust-lang/rust/commit/1eb2d44ddf07d6867e9ed91723de1276e029673b/checks/47347895617/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
.......... (130/140)
.........F (140/140)


/checkout/tests/rustdoc-gui/notable-trait.goml notable-trait... FAILED
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 11
    from `tests/rustdoc-gui/notable-trait.goml` line 38
    from `tests/rustdoc-gui/notable-trait.goml` line 64: different Y values: 332.5 != 331.5: for command `compare-elements-position: (
            "//*[@id='method.create_an_iterator_from_read']//a[normalize-space()='NotableStructWithLongName']",
            "//*[@id='method.create_an_iterator_from_read']//*[@class='tooltip']",
            ["y"],
        )`
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 11
    from `tests/rustdoc-gui/notable-trait.goml` line 80: different Y values: 356.5 != 355.5: for command `compare-elements-position: (
            "//*[@id='method.create_an_iterator_from_read']//a[normalize-space()='NotableStructWithLongName']",
            "//*[@id='method.create_an_iterator_from_read']//*[@class='tooltip']",
            ["y"],
        )`
[ERROR] `tests/rustdoc-gui/notable-trait.goml` line 11
    from `tests/rustdoc-gui/notable-trait.goml` line 38
    from `tests/rustdoc-gui/notable-trait.goml` line 88: different Y values: 391.5 != 390.5: for command `compare-elements-position: (
            "//*[@id='method.create_an_iterator_from_read']//a[normalize-space()='NotableStructWithLongName']",
            "//*[@id='method.create_an_iterator_from_read']//*[@class='tooltip']",
            ["y"],
        )`

Error: ()
Build completed unsuccessfully in 0:04:42
  local time: Mon Aug  4 16:48:56 UTC 2025
```

</details>

--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -1838,6 +1838,10 @@ instead, we check that it's not a "finger" cursor.
 	border-right: 3px solid var(--target-border-color);
 }
 
+a.tooltip {
+	font-family: var(--font-family);
+}
+
 .code-header a.tooltip {
 	color: inherit;
 	margin-right: 15px;

--- a/tests/rustdoc-gui/notable-trait.goml
+++ b/tests/rustdoc-gui/notable-trait.goml
@@ -8,10 +8,10 @@ define-function: (
     [x, i_x],
     block {
         // Checking they have the same y position.
-        compare-elements-position: (
+        compare-elements-position-near: (
             "//*[@id='method.create_an_iterator_from_read']//a[normalize-space()='NotableStructWithLongName']",
             "//*[@id='method.create_an_iterator_from_read']//*[@class='tooltip']",
-            ["y"],
+            {"y": 1},
         )
         // Checking they don't have the same x position.
         compare-elements-position-false: (

--- a/tests/rustdoc-gui/src/test_docs/lib.rs
+++ b/tests/rustdoc-gui/src/test_docs/lib.rs
@@ -767,3 +767,17 @@ pub mod impls_indent {
         pub fn bar() {}
     }
 }
+
+pub mod tooltips {
+    pub struct X;
+
+    impl X {
+        pub fn bar() -> Vec<u8> {
+            Vec::new()
+        }
+    }
+
+    pub fn bar() -> Vec<u8> {
+        Vec::new()
+    }
+}

--- a/tests/rustdoc-gui/tooltips.goml
+++ b/tests/rustdoc-gui/tooltips.goml
@@ -1,0 +1,15 @@
+// This test checks that the right font is applied to the `i` tooltip element.
+
+define-function: (
+    "check-font",
+    [path],
+    block {
+        go-to: "file://" + |DOC_PATH| + "/test_docs/" + |path|
+        assert-css: (
+            "a.tooltip", {"font-family": '"Source Serif 4", NanumBarunGothic, serif'}, ALL,
+        )
+    }
+)
+
+call-function: ("check-font", {"path": "tooltips/fn.bar.html"})
+call-function: ("check-font", {"path": "tooltips/struct.X.html"})

